src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -498,6 +499,18 @@ private static string GenerateUpgradeReport(string projectName, string backupPat
                         uiService.Projects,
                         cancellationToken)).ToArray();
 
+                    var packageNamespacesConfiguration = PackageNamespacesConfiguration.GetPackageNamespacesConfiguration(uiService.Settings);
+                    bool areNamespacesEnabled = packageNamespacesConfiguration?.AreNamespacesEnabled ?? false;
```

Comment:
There is a lot of duplication here and all the package commands. 
You can consider a helper method.

Reply:
I created helper method for PMC usage case, but here usage is only 1 time, better to keep as it's.